### PR TITLE
chore: refactor node namespaces

### LINKED ISSUE #170: Refactor  `hardhat_`, `evm_`, and `zks_` namespace implementations
## 📝 **Description**

Refactor the `hardhat_`, `evm_`, and `zks_` namespace implementations for some functions to be stored in a centralized location, so they can be re-used accordingly. 

For example, the following endpoints are doing the same thing:
* `evm_setAccountNonce`
* `hardhat_setNonce`
* `anvil_setNonce`

## 🤔 **Rationale**

Centralizing namespaces would lead to a significantly lower effort for contributors implementing other namespaces and/or methods.

## 📋 Additional Context
This is a prerequisite for #160

### PR DESCRIPTION
**Pull request by nbaztec**
# What :computer: 
* Isolates the supported namespaces in `crate::namespaces`
* Implements the supported namespaces in `crate::node`
* Common functionality is implemented in module `crate::node::in_memory_ext`

# Why :hand:
* Easier to add new namespaces
* Allows sharing of common functionality across namespaces
* Less code since the namespaces are directly implemented on `InMemoryNode` like the `eth_`

# Evidence :camera:
Tests pass

# Notes :memo:
* Fixes #170 
* In case the `in_memory_ext` method name collides with a namespace name the alternative call syntax of rust is being used

```rust
// If `mine_block` is both part of ext methods and the evm namespace impl, then

// Instead of 
node.mine_block(1)

// Use
EvmNamespaceT::mine_block(&node, 1);

// for calling the evm method
```

### CODE REVIEW

File: src/node/in_memory_ext.rs [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    ///
+    /// # Parameters
+    /// - `time_delta`: The number of seconds to increase time by
+    ///
+    /// # Returns
+    /// The applied time delta to `current_timestamp` value for the InMemoryNodeInner.
+    pub fn increase_time(&self, time_delta_seconds: u64) -> Result<u64> {
+        self.get_inner()
+            .write()
+            .map_err(|err| anyhow!("failed acquiring lock: {:?}", err))
```
Comment (MexicanAce) [inline src/node/in_memory_ext.rs:33]: TIL about `anyhow!`

File: src/node/in_memory_ext.rs [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                            let mut bytes = vec![0u8; 32];
+                            entry.to_big_endian(&mut bytes);
+                            bytes.to_vec()
+                        })
+                        .collect(),
+                );
+                writer.fork_storage.set_value(code_key, hash);
+            })
+    }
+}
```
Comment (MexicanAce) [inline src/node/in_memory_ext.rs:321]: Should we also be moving the unit tests into this file? Feels like a majority of the business logic is here

File: src/namespaces/configuration_api.rs [hunk 3]
```diff
@@ -0,0 +1,93 @@
+use crate::namespaces::Result;
+use crate::observability::LogLevel;
+use jsonrpc_derive::rpc;
+
+#[rpc]
+pub trait ConfigurationApiNamespaceT {
```
[the remark this answers is not in the document]
Comment (nbaztec) [reply src/namespaces/configuration_api.rs:6]: I thought about it, and I think yes it makes sense 😄

File: src/main.rs [hunk 4]
```diff
@@ -103,29 +93,24 @@ pub const RICH_WALLETS: [(&str, &str); 10] = [
...
-        io.extend_with(node.to_delegate());
-        io.extend_with(net.to_delegate());
-        io.extend_with(config_api.to_delegate());
-        io.extend_with(evm.to_delegate());
-        io.extend_with(zks.to_delegate());
-        io.extend_with(hardhat.to_delegate());
-        io.extend_with(debug.to_delegate());
+
+        io.extend_with(NetNamespaceT::to_delegate(node.clone()));
+        io.extend_with(ConfigurationApiNamespaceT::to_delegate(node.clone()));
```
[the remark this answers is not in the document]
Comment (nbaztec) [reply src/main.rs:108]: They all share the same inner node which is arc cloned so yeah that should be fine

File: src/main.rs [hunk 5]
```diff
@@ -103,29 +93,24 @@ pub const RICH_WALLETS: [(&str, &str); 10] = [
...
+        io.extend_with(DebugNamespaceT::to_delegate(node.clone()));
+        io.extend_with(EthNamespaceT::to_delegate(node.clone()));
+        io.extend_with(EvmNamespaceT::to_delegate(node.clone()));
+        io.extend_with(HardhatNamespaceT::to_delegate(node.clone()));
+        io.extend_with(ZksNamespaceT::to_delegate(node));
```
[the remark this answers is not in the document]
Comment (nbaztec) [reply src/main.rs:113]: `node` isn't being used after this statement that's why no need to clone

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**